src/Authentication/Authentication/Cmdlets/InvokeGraphRequest.cs

```diff
@@ -0,0 +1,159 @@
+
+    using Newtonsoft.Json;
+    using Microsoft.Graph.PowerShell.Authentication.Helpers;
+
+
+    [Cmdlet(VerbsLifecycle.Invoke, "GraphRequest", DefaultParameterSetName = Constants.UserParameterSet)]
+    [OutputType(typeof(Hashtable))]
+    public class InvokeGraphRequest : PSCmdlet
+    {
+        /// <summary>
+        /// Contains the values of HttpVerbs
+        /// </summary>
+        public enum HttpVerb
+        {
+            GET,
+            POST,
+            PUT,
+            PATCH,
+            DELETE
+        }
+        private readonly HttpClient _client;
+        public InvokeGraphRequest()
+        {
+            this.Method = HttpVerb.GET;
+            var authConfig = GraphSession.Instance.AuthContext;
+            this._client = HttpHelpers.GetGraphHttpClient(authConfig);
+        }
+
+        /// <summary>
+        /// Http Method
+        /// </summary>
+        [Parameter(ParameterSetName = Constants.UserParameterSet, Position = 1, Mandatory = true)]
+        public HttpVerb Method { get; set; }
+
+        /// <summary>
+        /// Uri to call using the Graph HttpClient can be segments such as /beta/me
+        /// or fully qualified url such as https://graph.microsoft.com/beta/me
+        /// </summary>
+        [Parameter(ParameterSetName = Constants.UserParameterSet, Position = 2, Mandatory = true)]
+        public Uri Uri { get; set; }
+
+        /// <summary>
+        /// Optional Http Body
+        /// </summary>
+        [Parameter(ParameterSetName = Constants.UserParameterSet, Position = 3)]
+        public object Body { get; set; }
+
+        protected override void BeginProcessing()
+        {
+            ValidateParameters();
+            base.BeginProcessing();
+        }
+
+        protected override void ProcessRecord()
+        {
+            base.ProcessRecord();
+            var authConfig = GraphSession.Instance.AuthContext;
+            var client = HttpHelpers.GetGraphHttpClient(authConfig);
+            try
+            {
+                var httpMethod = GetHttpMethod();
+                var requestMessage = new HttpRequestMessage(httpMethod, this.Uri)
+                {
+                    Content = new StringContent(null != Body ? JsonConvert.SerializeObject(Body) : @"{}", Encoding.UTF8,
+                        CoreConstants.MimeTypeNames.Application.Json)
+                };
+                var responseMessage = client.SendAsync(requestMessage).GetAwaiter().GetResult();
+                var responseString = responseMessage.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+                var result = JsonConvert.DeserializeObject<Hashtable>(responseString);
+                if (responseMessage.IsSuccessStatusCode)
+                {
+                    WriteObject(result);
+                }
+                else
+                {
+                    WriteError(new ErrorRecord(new Exception(responseString), Guid.NewGuid().ToString(),
```

Comment:
It would also be useful to write out the response headers in case !responseMessage.IsSuccessStatusCode

Reply:
![image](https://user-images.githubusercontent.com/1641829/84331058-ddeb1480-ab91-11ea-88a0-20248216a874.png)

@MIchaelMainer I am considering this kind of output